build/hwloc.m4

```diff
@@ -0,0 +1,105 @@
+dnl -------------------------------------------------------- -*- autoconf -*-
+dnl Licensed to the Apache Software Foundation (ASF) under one or more
+dnl contributor license agreements.  See the NOTICE file distributed with
+dnl this work for additional information regarding copyright ownership.
+dnl The ASF licenses this file to You under the Apache License, Version 2.0
+dnl (the "License"); you may not use this file except in compliance with
+dnl the License.  You may obtain a copy of the License at
+dnl
+dnl     http://www.apache.org/licenses/LICENSE-2.0
+dnl
+dnl Unless required by applicable law or agreed to in writing, software
+dnl distributed under the License is distributed on an "AS IS" BASIS,
+dnl WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+dnl See the License for the specific language governing permissions and
+dnl limitations under the License.
+
+dnl
+dnl hwloc.m4: Trafficserver's hwloc autoconf macros
+dnl
+
+dnl
+dnl TS_CHECK_HWLOC: look for hwloc libraries and headers
+dnl
+AC_DEFUN([TS_CHECK_HWLOC], [
```

Comment:
Normally the macro would just do the check, taking arguments of what to do if the check succeeded or failed. The caller would be responsible for throwing the error.